shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/visitor/impl/MySQLDALVisitor.java

```diff
@@ -145,14 +150,32 @@ public ASTNode visitShowOther(final ShowOtherContext ctx) {
     public ASTNode visitSetVariable(final SetVariableContext ctx) {
         SetStatement result = new SetStatement();
         if (null != ctx.variable()) {
-            result.setVariable((VariableSegment) visit(ctx.variable()));
+            List<VariableContext> variableContextList = ctx.variable();
+            List<VariableSegment> variableSegmentList = new ArrayList<VariableSegment>();
+            for (VariableContext variableContext : variableContextList) {
+                variableSegmentList.add((VariableSegment) visit(variableContext));
+            }
+            result.setVariable(variableSegmentList.get(0));
+            
+            List<VariableValueContext> variableValueContextList = ctx.variableValue();
+            List<VariableValueSegment> variableValueSegmentList = new ArrayList<VariableValueSegment>();
+            for (VariableValueContext variableValueContext : variableValueContextList) {
+                variableValueSegmentList.add((VariableValueSegment) visit(variableValueContext));
+            }
+            result.setVariableValue(variableValueSegmentList.get(0));
```

Comment:
Is it correct for setVariableValue to variableValueSegmentList.get(0)?

Reply:
re-design